persistence/src/commonMain/db_user/com/wire/kalium/persistence/ConversationDetails.sq

```diff
@@ -158,3 +158,7 @@ SELECT * FROM ConversationDetails WHERE qualifiedId = ?;
 
 selectConversationDetailsByGroupId:
 SELECT * FROM ConversationDetails WHERE mls_group_id = ?;
+
+selectActiveOneOnOneConversationDetails:
+SELECT * FROM ConversationDetails
+WHERE qualifiedId = (SELECT active_one_on_one_conversation_id FROM User WHERE qualified_id = :user_id);
```

Comment:
I'd recommend using a JOIN instead of a subquery, it is more efficient.
```
SELECT ConversationDetails.*
FROM ConversationDetails
JOIN User ON ConversationDetails.qualifiedId = User.active_one_on_one_conversation_id
WHERE User.qualified_id = :user_id;
```

Reply:
I analyzed the performance of both queries using EXPLAIN ANALYZE in SQLite. The execution plans are identical, both queries utilize the same indexes and have no significant performance difference.

Since there's no additional cost for the subquery, the choice between JOIN and WHERE (SELECT …) comes down to readability and preference.

Reply:
ok.. thanks for checking